### PR TITLE
43 validate nominate

### PR DESCRIPTION
**Pull request by ann-kilzer**
Part two of #43 

Validation for the Speaker Nomination page.

We now validate
- English Name
- Japanese Name
- Email
- Bio
- Submitter name
- Submitter email
- consent

Additional work in this PR: 
- unit tests for Japanese validator
- Wire up the submit button

Screenshots:
web

web

Remaining work (for a separate PR to minimize merge conflicts)
- ensure English name uses Latin characters (be sure to support accents and foreign names, perhaps non-japanese is a better check)
- Check city and prefecture
- Validate any new fields added in #50 
- make the email validation chill a bit, shouldn't turn red after the first character entered, that's aggro

### CODE REVIEW

File: web/src/components/nomination/NameInput.vue [hunk 1]
```diff
@@ -7,10 +7,12 @@
     >
       <v-text-field
         ref="english"
-        :value="value.title"
+        :value="value.english"
```
Comment (tuttiq) [inline web/src/components/nomination/NameInput.vue:10]: we're gonna have a few conflicts on these

File: web/src/components/nomination/NominateForm.vue [hunk 2]
```diff
@@ -1,12 +1,22 @@
 <template>
-  <v-form class="my-3">
+  <v-form
+    class="my-3"
+    @submit.prevent="handleSubmit"
```
Comment (tuttiq) [inline web/src/components/nomination/NominateForm.vue:4]: I implemented this on my PR as well after your review comment

File: web/src/components/nomination/NominateForm.vue [hunk 3]
```diff
@@ -51,14 +79,37 @@ export default {
...
   data() {
     return {
       form: {
         speakerName: {
-          title: '',
-          first: '',
-          middle: '',
-          last: '',
+          english: '',
+          japanese: '',
```
Comment (tuttiq) [inline web/src/components/nomination/NominateForm.vue:112]: same here, I changed the whole structure of the data store to make it easier to parse for Airtable. ごめんなさい 😭

File: web/src/components/nomination/NominateForm.vue [hunk 4]
```diff
@@ -79,8 +130,63 @@ export default {
...
+      this.$set(this.form, 'speakerBio', '');
+      this.$set(this.form, 'location', { city: '', prefecture: '' });
+      this.$set(this.form, 'submitterInput', { name: '', email: '' });
+      this.$v.$reset();
+    },
+    handleSubmit() {
+      this.$v.$touch();
+      if (this.$v.$invalid) {
+        console.error('invalid form');
+      }
```
Comment (tuttiq) [inline web/src/components/nomination/NominateForm.vue:189]: this should be easy to merge on the top of what I implemented for this method

File: web/src/validators/japanese.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+// custom vuelidate validator to ensure input is Japanese
+// inspired by
```
Comment (tuttiq) [inline web/src/validators/japanese.js:2]: inspired by...? lol

File: web/src/validators/japanese.js [hunk 6]
```diff
@@ -0,0 +1,9 @@
...
+const japanese = (value) => {
+  const jaRegex = /^[\u3000-\u303f\u3040-\u309f\u30a0-\u30ff\uff00-\uff9f\u4e00-\u9faf\u3400-\u4dbf]*$/;
```
Comment (tuttiq) [inline web/src/validators/japanese.js:4]: that's pretty cool

File: web/tests/unit/validator-japanese.spec.js [hunk 7]
```diff
@@ -0,0 +1,23 @@
+/* eslint-disable no-unused-expressions */
```
Comment (tuttiq) [inline web/tests/unit/validator-japanese.spec.js:1]: YAY TESTS! Thanks for starting this! ❤️

File: web/src/validators/japanese.js (hunk 5 above)
Comment (ann-kilzer) [reply web/src/validators/japanese.js:2]: forgot to paste the link...

**Pull request merged**